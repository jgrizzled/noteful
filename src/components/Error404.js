import React from 'react';
import ErrorMessage from 'components/common/ErrorMessage';

const Error404 = () => <ErrorMessage>Error: page not found</ErrorMessage>;

export default Error404;
