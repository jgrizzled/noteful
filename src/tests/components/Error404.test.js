import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Error404 from 'components/Error404';

const error404 = <Error404 />;

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(error404, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', () => {
  const tree = renderer.create(error404).toJSON();
  expect(tree).toMatchSnapshot();
});
