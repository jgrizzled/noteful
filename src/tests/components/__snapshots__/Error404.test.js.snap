// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<h1
  className="sc-AykKC hpENYW"
>
  Error: page not found
</h1>
`;
